### ISSUE TITLE
Migrate user profile

### ISSUE THREAD

**Issue by githubbob42** [githubbob42/mingle2github2#4249]
**[Mingle Card: 4585](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/4585)**
Narrative
---------

In the effort to eliminate alpine-fetch, we need to move the user profile sync process to alpine-mobile and remove it from alpine-fetch

Acceptance Criteria
-------------------

_Make alpine-mobile sync_ user profile _data and remove the_ user profile _data from fetch.  It should continue to pass all tests._

Analysis
--------

Today, \`steps/user-profile\` provides a blob of data that is a composite of user, contact, FX5\_\_FX\_Settings\_\_c, the user’s identity token (oauth instance\_url) and some environmental configuration settings (log entries key).

We need to identify where each of these pieces are used in mobile so we can devise an alternative approach to accessing this data. This approach will probably rely on the following:

*   The User table is already included in the incremental sync
*   The Contact table is already included in the incremental sync
*   The Sync already calls a custom \`/getIdentity\` endpoint before starting the rest of the sync process (this endpoint already returns the \`instance\_url\`, and identifier for the current logged in user and the log entries key)
*   We can add FX5\_\_FX\_Settings\_\_c as a table in the sync if we need access to custom setttings data

​Identity

We currently call the \`/getIdentity\` endpoint for every sync. We should save the response for this call to a location other than the store key for “user-profile” (perhaps cache storage). This will help us differentiate between what we consider to be the identity response and the user table data.

Eventually, we should replace the \`/getIdentity\` endpoint with a call to \`/services/data/v37.0\`, followed up with a call to the url returned on the \`identity\` key on the previous response payload. 

​Custom Settings

We use custom settings in two locations in mobile:

*   bindings/inline-help.js
*   bindings/auto-select.js

We should be able to update these two code paths to query the FX5\_\_FX\_Settings\_\_c table for custom data, rather than grabbing the setting from the user-profile.

UPDATE: since we’re unable to apply sharing rules to the FX\_Settings table, we’ll need to consider how to filter this data. Going to split custom settings into a separate card (#4606) in order to avoid stacking up too much work.

Config Keys (such as log entries id)

We use config keys in the following locations:

*   services/mixpanel.js
*   services/raygun.js
*   system/logger.js

If we store the config keys independent from the “user-profile”, we can should be able to update these code paths to no longer be dependent on the user-profile sync step (they’ll instead be dependent on \`/getIdentity\`). 

User

We use the user data from the user-profile step in the following locations in mobile:

*   print/helpers/user-profile
*   salesforce-data-context
*   templated\\filters\\timezone
*   templated\\filters\\locale

Both timezone and locale code paths can be updated to depend on the identity response from cache storage (rather than user profile). 

The \`$User\` property on salesforce-data-context will have to be updated to issue a query on the User table, using the user\_id value from /getIdentity.

Lastly, print/helpers/user-profile will have to be updated to issue a query on the User table, using the user\_id value from /getIdentity, followed by a query to the Contact table (in order to retain backwards-compatibility with existing report templates, we’ll likely still need to merge these two entities together).

Contact

We use the contact data from the user-profile step in the following locations in mobile:

*   print/helpers/user-profile

*   store/ticket-item-picker

As per the previous section, print/helpers/user-profile will have to be updated to issue a query on the User table, using the user\_id value from /getIdentity, followed by a query to the Contact table (in order to retain backwards-compatibility with existing report templates, we’ll likely still need to merge these two entities together).

Equally, store/ticket-item-picker (which uses the current user’s contact record to identify the name of their office), will have to be updated to issue queries for user, then contact, then office. 

\---------------

%sSyncVersion\_\_c, (via getSyncVersion() in alpine-server/sforce/index.js)

autopopulate\_required\_ticket\_item\_lookup\_\_c, Force\_Inline\_Help\_\_c ( alpine-fetch/sync/steps/userprofile.js)

/services/data/v37.0/query/?q=SELECT+SyncVersion\_\_c,autopopulate\_required\_ticket\_item\_lookup\_\_c,Force\_Inline\_Help\_\_c+FROM+FX5\_\_FX\_Settings\_\_c+WHERE+(SetupOwnerId+=+'005i0000002XxGdAAK'+OR+SetupOwnerID+=+'00ei0000000f2aXAAQ'+OR+(NOT+Name+like+'%25FX+Settings%25'))+AND+IsDeleted+=+False+order+by+name+desc+LIMIT+1

Related Stories
---------------

Tasks
-----

{{ table query: SELECT Number, Name, Owner, 'Task Status' WHERE Type = Task and Story = THIS CARD ORDER BY Status }}

Defects
-------

{{ table query: SELECT Number, Name, Owner, 'Status' WHERE Type = Defect and 'Related Story' = THIS CARD }}

Test Plan
---------

Update Docs: As per any other table in the sync, we need to make sure FX\_Settings can be filtered appropriately in order to ensure we don’t cause unnecessary overhead when syncing.

We user user data in the following locations in mobile: We user user data in the following locations in mobile:

**Comment by githubbob42** [comment]
### Feature: #4177 Move the remaining parts of fetch to Sync 4 ([Mingle](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/4512))

**Comment by githubbob42** [comment]
**Pull Request #2112**
